### ISSUE TITLE
[Research] Add EEG+Voice Transformer Fusion (TMNet Architecture)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#21]
## What

Implement a **Transformer-based EEG+Voice fusion model** inspired by TMNet (ICT Express, May 2025) — the first dedicated EEG+Speech multimodal framework. Replace the current weighted-average fusion in `multimodal_emotion_fusion.py` with a trainable cross-modal attention mechanism.

## Why

- Current fusion: simple weighted average of EEG and voice emotion probabilities
- TMNet achieves **98.89% accuracy** (within-subject, controlled) using Transformer cross-attention
- EEG and voice signals capture **complementary** emotional information:
  - EEG: real-time neural correlates (valence via FAA, arousal via beta/alpha)
  - Voice: prosodic/acoustic markers (pitch, energy, speech rate, formants)
- Neurocomputing Feb 2026 review confirms: multimodal EEG+speech fusion consistently outperforms unimodal
- Interspeech 2025 challenge: graph-based multimodal fusion achieved new SOTA

### Step 1: Implement dual-branch feature extractors
```python
# ml/models/eeg_voice_fusion.py

class EEGBranch(nn.Module):
    """CNN-BiLSTM for EEG temporal features.
    Input: (batch, 4_channels, time_samples) from Muse 2
    Output: (batch, seq_len, eeg_embed_dim)
    """

class VoiceBranch(nn.Module):
    """BiGRU for speech features.
    Input: (batch, time_frames, voice_features) from emotion2vec+ or eGeMAPS
    Output: (batch, seq_len, voice_embed_dim)
    """

class TransformerFusion(nn.Module):
    """Cross-modal Transformer attention.
    Aligns temporal dynamics between EEG and speech.
    Input: EEG embeddings + Voice embeddings
    Output: fused emotion prediction (6-class + valence + arousal)
    """
```

### Step 2: Data collection pipeline
- During "voice watch" EEG recording sessions, capture both EEG + audio simultaneously
- Store paired (eeg_epoch, audio_segment) with timestamps
- Use existing emotion2vec+ features as voice input
- Use existing 41-feature multichannel EEG features as EEG input

### Step 3: Training
- Pre-train branches separately on available data
- Joint fine-tuning with cross-attention fusion
- Use contrastive loss to align EEG and voice representations in shared space

### Step 4: Integration
- Wire into `/analyze-eeg` endpoint when voice data is available
- Fall back to unimodal (EEG-only or voice-only) when one modality is missing
- Return `fusion_type: "transformer" | "weighted_avg" | "eeg_only" | "voice_only"` in response

### Files to Modify
- `ml/models/eeg_voice_fusion.py` (NEW) — Transformer fusion model
- `ml/models/multimodal_emotion_fusion.py` — Replace/augment weighted average
- `ml/api/routes.py` — Update `/analyze-eeg` to accept simultaneous voice+EEG
- `ml/training/train_fusion.py` (NEW) — Joint training script
- `client/src/pages/voice-watch.tsx` — Ensure EEG + audio captured simultaneously

### Expected Impact
- Multimodal emotion accuracy: **74% → 88-92%** (within-subject with fusion)
- Better cross-subject: **+8-12 points** over EEG-only

## Complexity
HIGH — requires paired EEG+voice data collection, custom PyTorch model, joint training

## References
- TMNet: https://www.sciencedirect.com/science/article/pii/S2405959525000517
- EEG+Speech fusion review (Neurocomputing 2026): https://www.sciencedirect.com/science/article/abs/pii/S0925231225029078
- Interspeech 2025 SER challenge: https://arxiv.org/abs/2506.02088
